crates/sb_ai/onnxruntime/mod.rs

```diff
@@ -3,19 +3,29 @@ pub(crate) mod onnx;
 pub(crate) mod session;
 mod tensor;
 
-use std::{borrow::Cow, collections::HashMap};
+use std::{borrow::Cow, cell::RefCell, collections::HashMap, rc::Rc, sync::Arc};
 
-use anyhow::Result;
-use deno_core::op2;
+use anyhow::{anyhow, Result};
+use deno_core::{op2, OpState};
 
 use model_session::{ModelInfo, ModelSession};
+use ort::Session;
 use tensor::{JsTensor, ToJsTensor};
 
 #[op2]
 #[to_v8]
-pub fn op_sb_ai_ort_init_session(#[buffer] model_bytes: &[u8]) -> Result<ModelInfo> {
+pub fn op_sb_ai_ort_init_session(
+    state: Rc<RefCell<OpState>>,
+    #[buffer] model_bytes: &[u8],
+) -> Result<ModelInfo> {
+    let mut state = state.borrow_mut();
     let model_info = ModelSession::from_bytes(model_bytes)?;
 
+    let mut sessions = { state.try_take::<Vec<Arc<Session>>>().unwrap_or_default() };
+
+    sessions.push(model_info.inner());
+    state.put(sessions);
+
```

Comment:
`try_take` is to remove a value from the container.
We want to keep the Vec<T> in state, so we shouldn't remove it.

Please check if the Vec<T> exists with the `has` method, and if not, put an empty Vec<T> with `put` and then add a session with `borrow_mut`.

```rust
    let mut state = state.borrow_mut();
    
    if !state.has::<Vec<meow>>() {
        state.put(Vec::<meow>::new())
    }
    
    let vec_mut = state.borrow_mut::<Vec<meow>>();
    vec_mut.push(model_info.inner());
```

Reply:
Ah, the code makes sense.
You don't need to fix it.